### ISSUE TITLE
IXMLService MD5 Crypto Breaks When FIPS Validated Algo Setting Enabled

### ISSUE THREAD

**Issue by jdkang** [chocolatey/choco#556]
The registered delegate for type IXmlService threw an exception. The registered delegate for type IHashProvider threw an exception. Exception has been thrown by the target of an invocation.   

verbose:

    Chocolatey had an error occur:
    SimpleInjector.ActivationException: The registered delegate for type IXmlService threw an exception. The registered delegate for type IHashProvider threw an exception. Exception has been thrown by the target of an invocation. ---> SimpleInjector.ActivationException: The registered delegate for type IHashProvider threw an exception. Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: This implementation is not part of the Windows Platform FIPS validated cryptographic algorithms.
       at System.Security.Cryptography.MD5CryptoServiceProvider..ctor()
       --- End of inner exception stack trace ---
       at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
       at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
       at System.Security.Cryptography.CryptoConfig.CreateFromName(String name, Object[] args)
       at System.Security.Cryptography.MD5.Create(String algName)
       at chocolatey.infrastructure.cryptography.CrytpoHashProvider..ctor(IFileSystem fileSystem, CryptoHashProviderType providerType)
       at chocolatey.infrastructure.app.registration.ContainerBinding.<>c__DisplayClass6.<RegisterComponents>b__2()
       at DynamicInstanceProducer3.GetInstance(Object[] constants)
       at SimpleInjector.CompilationHelpers.<>c__DisplayClass1b`1.<CompileInDynamicAssemblyAsClosure>b__1a()
       at SimpleInjector.Lifestyles.SingletonLifestyle.SingletonLifestyleRegistrationBase`1.CreateInstanceWithNullCheck()
       at System.Lazy`1.CreateValue()
       at System.Lazy`1.LazyInitValue()
       at SimpleInjector.Lifestyles.SingletonLifestyle.SingletonLifestyleRegistrationBase`1.BuildExpression()
       at SimpleInjector.Registration.BuildExpression(InstanceProducer producer)
       at SimpleInjector.InstanceProducer.BuildExpressionInternal()
       at System.Lazy`1.CreateValue()
       at System.Lazy`1.LazyInitValue()
       at SimpleInjector.InstanceProducer.BuildExpression()
       --- End of inner exception stack trace ---
       at SimpleInjector.InstanceProducer.BuildExpression()
       at SimpleInjector.Registration.BuildParameterExpressionFor(ParameterInfo parameter)
       at SimpleInjector.Registration.BuildConstructorParameters(ConstructorInfo constructor)
       at SimpleInjector.Registration.BuildNewExpression(Type serviceType, Type implementationType)
       at SimpleInjector.Registration.BuildTransientExpression[TService,TImplementation]()
       at SimpleInjector.Registration.BuildTransientDelegate[TService,TImplementation]()
       at SimpleInjector.Lifestyles.SingletonLifestyle.SingletonLifestyleRegistration`2.CreateInstance()
       at SimpleInjector.Lifestyles.SingletonLifestyle.SingletonLifestyleRegistrationBase`1.CreateInstanceWithNullCheck()
       at System.Lazy`1.CreateValue()
       at System.Lazy`1.LazyInitValue()
       at SimpleInjector.Lifestyles.SingletonLifestyle.SingletonLifestyleRegistrationBase`1.BuildExpression()
       at SimpleInjector.Registration.BuildExpression(InstanceProducer producer)
       at SimpleInjector.InstanceProducer.BuildExpressionInternal()
       at System.Lazy`1.CreateValue()
       at System.Lazy`1.LazyInitValue()
       at SimpleInjector.InstanceProducer.BuildInstanceCreator()
       at SimpleInjector.InstanceProducer.GetInstance()
       --- End of inner exception stack trace ---
       at SimpleInjector.InstanceProducer.GetInstance()
       at SimpleInjector.Container.GetInstance[TService]()
       at chocolatey.console.Program.Main(String[] args)

Caused by "FIPS Validation" being enabled on the system.

You can validate it with the following registry setting:

```powershell
function Disable-FipsAlgoRestriction {
    $currentVal = (Get-ItemProperty HKLM:\SYSTEM\CurrentControlSet\Control\Lsa\FipsAlgorithmPolicy -name Enabled).Enabled
    write-host "Current Value is $currentVal" -f green
    write-host "Overriding to value 0" -f green
    set-itemproperty HKLM:\SYSTEM\CurrentControlSet\Control\Lsa\FipsAlgorithmPolicy -name Enabled -Value 0
}
```

**Comment by gep13** [comment]
@jdkang I think this is duplicate of this:

https://github.com/chocolatey/choco/issues/540

Right?

**Comment by ferventcoder** [comment]
It is. Closed as a duplicate of #446

**Comment by ferventcoder** [comment]
@jdkang please subscribe/comment on #446.
